mfs/dir.go

```diff
@@ -131,7 +131,7 @@ func (d *Directory) cacheNode(name string, nd *dag.Node) (FSNode, error) {
 		ndir := NewDirectory(d.ctx, name, nd, d, d.dserv)
 		d.childDirs[name] = ndir
 		return ndir, nil
-	case ufspb.Data_File, ufspb.Data_Raw:
+	case ufspb.Data_File, ufspb.Data_Raw, ufspb.Data_Symlink:
```

Comment:
What the old code in AddChild did was check if `len(links) == 0`, and if so, cached it as a file. it didn't check type (symlink, file, etc) *at all* like `cacheNode` does.

This is because `ipfs add` uses the `mfs` package to add files, which means in order to handle symlinks in directories (like we do today on `ipfs add`), `AddChild` needs to also support symlinks.